### PR TITLE
Add sync_tensorboard telemetry

### PR DESCRIPTION
**Pull request by raubitsj**
Fixes WB-8615

Description
-----------
Adds telemetry for sync_tensorboard and wandb.tensorboard.patch()
Bonus: Cleans up early telemetry before a run is created.

Todo:
- [x] sync_tensorboard test
- [x] wandb.tensboard.patch test

Testing
-------
Basic yea tests

Checklist
-------
- Name PR "[WB-NNNN][WB-MMMM] Add support for..." similar to entries in CHANGELOG.md
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

File: wandb/sdk/wandb_init.py [hunk 1]
```diff
@@ -660,15 +657,15 @@ def init(self) -> Union[Run, RunDisabled, None]:  # noqa: C901
         return run
 
     def _set_init_telemetry_attrs(self, kwargs):
```
Comment (kptkin) [inline wandb/sdk/wandb_init.py:659]: nit: maybe we can remove this function and just add the values directly to the `_init_telemetry_obj`?

File: wandb/sdk/lib/telemetry.py [hunk 2]
```diff
@@ -40,8 +42,10 @@ def __exit__(
         self._run._telemetry_callback(self._obj)
 
 
-def context(run: "wandb_run.Run" = None) -> ContextManager[TelemetryRecord]:
-    return _TelemetryObject(run=run)
+def context(
+    run: "wandb_run.Run" = None, obj: TelemetryRecord = None
```
Comment (kptkin) [inline wandb/sdk/lib/telemetry.py:46]:
```suggestion
    run: Optional["wandb_run.Run"] = None, obj: Optional[TelemetryRecord] = None
```

File: wandb/sdk/lib/telemetry.py [hunk 3]
```diff
@@ -22,9 +22,11 @@
 class _TelemetryObject(object):
     _run: Optional["wandb_run.Run"]
 
-    def __init__(self, run: "wandb_run.Run" = None) -> None:
+    def __init__(
+        self, run: "wandb_run.Run" = None, obj: TelemetryRecord = None
```
Comment (kptkin) [inline wandb/sdk/lib/telemetry.py:26]:
```suggestion
        self, run: Optional["wandb_run.Run"] = None, obj: Optional[TelemetryRecord] = None
```

File: wandb/sdk/lib/telemetry.py [hunk 4]
```diff
@@ -22,9 +22,11 @@
 class _TelemetryObject(object):
     _run: Optional["wandb_run.Run"]
```
Comment (kptkin) [inline wandb/sdk/lib/telemetry.py:23]: should we add `_obj` here?

File: wandb/sdk/wandb_init.py (hunk 1 above)
Comment (raubitsj) [reply wandb/sdk/wandb_init.py]: i dont love this, but i dislike how cluttered _setup() is already also.  but given the current state, i agree,  i moved it into the setup function.

File: wandb/sdk/lib/telemetry.py (hunk 3 above)
Comment (raubitsj) [reply wandb/sdk/lib/telemetry.py:26]: i know this is the preferred style in pep now.. but i didnt know it when i wrote most of the code... lets turn on the mypy check and change this all at once.. there is a flag for it i recall.
We do implecit optionals throughout the library.

**Review by kptkin — APPROVED** [review]
Nice! :)

File: wandb/sdk/lib/telemetry.py (hunk 4 above)
Comment (raubitsj) [reply wandb/sdk/lib/telemetry.py:23]: yes! fixed

File: wandb/sdk/wandb_init.py [hunk 5]
```diff
@@ -85,13 +87,10 @@ def __init__(self):
         self._wl = None
         self._reporter = None
         self._use_sagemaker = None
```
Comment (kptkin) [inline wandb/sdk/wandb_init.py:89]: this one can also be folded into the `_init_telemetry_obj` solution

Comment (raubitsj) [reply wandb/sdk/wandb_init.py:89]: oh,, thanks. i missed that one, fixing.

**Pull request merged**